packages/neuron-wallet/src/services/tx/transaction-service.ts

```diff
@@ -358,20 +358,39 @@ export class TransactionsService {
         const value = sums.get(tx.hash!) || BigInt(0)
 
         let typeArgs: string | undefined | null
+        let txType: string = ''
+        let assetAccountType: AssetAccountType | string = ''
+
         const sudtInput = anyoneCanPayInputs.find(
           i => i.transactionHash === tx.hash && assetAccountInfo.isSudtScript(i.typeScript()!)
         )
+        const sudtOutput = anyoneCanPayOutputs.find(
+          o => o.outPointTxHash === tx.hash && assetAccountInfo.isSudtScript(o.typeScript()!)
+        )
         if (sudtInput) {
           typeArgs = sudtInput.typeArgs
+          if (!sudtOutput) {
+            txType = 'destroy'
+            assetAccountType = AssetAccountType.SUDT
+          }
         } else {
-          const sudtOutput = anyoneCanPayOutputs.find(
-            o => o.outPointTxHash === tx.hash && assetAccountInfo.isSudtScript(o.typeScript()!)
-          )
           if (sudtOutput) {
             typeArgs = sudtOutput.typeArgs
+            txType = 'create'
+            assetAccountType = AssetAccountType.SUDT
           }
         }
 
+        const ckbInput = anyoneCanPayInputs.find(i => i.transactionHash === tx.hash && !i.typeScript())
```

Comment:
`anyoneCanPayInputs` must include `type script`, because it's searching conditions include  `input.typeHash IS NOT NULL`.https://github.com/nervosnetwork/neuron/blob/50e112e67b9de1e679a8e8fd5a4a68c12557f17b/packages/neuron-wallet/src/services/tx/transaction-service.ts#L259

Reply:
Just to clarify, I use `anyoneCanPayInputs` because @Keith-CY's [comment](https://github.com/Magickbase/neuron-public-issues/issues/10#issuecomment-1107784182) says that 

> CKB Asset Account's type script is null, so anyone-can-pay input could have no type scripts.

Maybe I misunderstood this part sentence..😅

So maybe what I should do is to add a query like this:

```js
const assetAccountInputs = await connection
      .getRepository(InputEntity)
      .createQueryBuilder('input')
      .where(
        `
        input.transactionHash IN (:...txHashes) AND
        input.lockArgs in (select publicKeyInBlake160 from hd_public_key_info where walletId = :walletId) AND
        input.lockCodeHash = :lockCodeHash`,
        {
          txHashes,
          walletId: params.walletID,
          lockCodeHash: assetAccountInfo.anyoneCanPayCodeHash
        }
      )
      .getMany()
// then
const anyoneCanPayInputs = assetAccountInputs.filter(i => i.typeScript()) // so it has a typescript
const ckbAssetInput = assetAccountInputs.filter(i => !i.typeScript())

// similar for outputs
```

Is this correct @yanguoyu  @Keith-CY?

Reply:
> Just to clarify, I use `anyoneCanPayInputs` because @Keith-CY's [comment](https://github.com/Magickbase/neuron-public-issues/issues/10#issuecomment-1107784182) says that
> 
> > CKB Asset Account's type script is null, so anyone-can-pay input could have no type scripts.
> 
> Maybe I misunderstood this part sentence..😅
> 
> So maybe what I should do is to add a query like this:
> 
> ```js
> const assetAccountInputs = await connection
>       .getRepository(InputEntity)
>       .createQueryBuilder('input')
>       .where(
>         `
>         input.transactionHash IN (:...txHashes) AND
>         input.lockArgs in (select publicKeyInBlake160 from hd_public_key_info where walletId = :walletId) AND
>         input.lockCodeHash = :lockCodeHash`,
>         {
>           txHashes,
>           walletId: params.walletID,
>           lockCodeHash: assetAccountInfo.anyoneCanPayCodeHash
>         }
>       )
>       .getMany()
> // then
> const anyoneCanPayInputs = assetAccountInputs.filter(i => i.typeScript()) // so it has a typescript
> const ckbAssetInput = assetAccountInputs.filter(i => !i.typeScript())
> 
> // similar for outputs
> ```
> 
> Is this correct @yanguoyu @Keith-CY?

Exactly, the previous select clause only gets sUDT Asset Account Input/Output since it requires `type` not to be empty.

In this PR, we're going to get sUDT/CKB Asset Account, so the condition `type hash is not null` should be eliminated.